### PR TITLE
migrate odata-mock to typescript

### PR DESCRIPTION
**Pull request by atakanyenel**
There are many `any` notations, which is non-type in typescript. I will remove them step by step in this PR

### CODE REVIEW

**Comment by atakanyenel** [comment]
I also realized that we are not using `defaultConfig.json`. Should it be deleted ?

File: odata-mock/src/server/types.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+
+type apiType = {
+    name: string,
+    metadata: string,
+    specification: string,
+    type: string
+}
+type VarkesConfigType = {
+    name: string,
+    apis: apiType[]
```
Comment (obadrawi) [inline odata-mock/src/server/types.ts]: add events also to VarkesConfigType
Also if one of the elements don't exist in the file will that cause a problem?

File: odata-mock/src/server/server.ts [hunk 2]
```diff
@@ -1,19 +1,18 @@
...
-
-var configPath
+import { logger as LOGGER } from "./logger"
 
 var runAsync = async () => {
+    var configPath: string = ""
     if (process.argv.length > 2) {
-        configPath = process.argv[2]
+
+        var configPath = process.argv[2]
```
Comment (obadrawi) [inline odata-mock/src/server/server.ts]: wouldn't the second configPath be local to the if only like this?

File: odata-mock/package.json [hunk 3]
```diff
@@ -1,14 +1,17 @@
...
-  "main": "server/app.js",
+  "main": "dist/server/app.js",
+  "types": "dist/app.d.ts",
   "engines": {
     "node": ">=4"
   },
   "scripts": {
-    "start": "node server/server",
+    "compile": "tsc && cd src && copyfiles test/services.xml ../dist/ && copyfiles test/varkes_config.json ../dist/ && copyfiles server/resources/* ../dist/",
+    "start": "node dist/server/server.js",
```
Comment (obadrawi) [inline odata-mock/package.json]: could you add the config file path to the start

File: odata-mock/src/server/server.ts (hunk 2 above)
Comment (atakanyenel) [reply odata-mock/src/server/server.ts]: yep ,fixed

File: odata-mock/src/server/types.ts (hunk 1 above)
Comment (atakanyenel) [reply odata-mock/src/server/types.ts]: added

**Pull request merged**